data/alsa_drivers.rb

```diff
@@ -107,15 +121,12 @@ def params
 
   # find all sound drivers below the given path
   def self.find_all(path)
-    ret = []
-    lst = `find #{path} -type f -name 'snd-*.ko'`.split("\n").sort{|p1, p2| 
-	p1.split('/').last <=> p2.split('/').last
-    }
+    result = []
 
-    lst.each do |m|
-      ret << AlsaModule.new(m)
-    end
+    output = Command.output("find", path, "-type", "f", "-name", "snd-*.ko")
```

Comment:
Come to think about it, couldn't that simply be replaced by Ruby `'Find.find()` and one simple filter iterator? Do we need to call an external program here at all? I think Ruby can handle this just fine.

We don't even need to consider a chroot environment here; this module will clearly never be run in the first part of the installation where this would matter at all.

Reply:
Yes, maybe we can use "plain" ruby here. I will try it.